src/lib/components/Header/Nav/NavItem.test.ts

```diff
@@ -0,0 +1,109 @@
+import "@testing-library/jest-dom";
+import { render, screen } from "@testing-library/svelte";
+import userEvent from "@testing-library/user-event";
+import { describe, it, expect } from "vitest";
+
+import NavItem from "./NavItem.svelte";
+import TestSlotWrapper from "$lib/components/TestSlotWrapper.svelte";
+
+const generateMenuItems = (n: number) =>
+  [...Array(n)].map((_, i) => ({
+    id: `${i + 1}`,
+    name: `nav item ${i + 1}`,
+    link: `/${i + 1}`,
+  }));
```

Comment:
This is really similar to the `generateMenuItems` function in `Header.stories.ts`; I wasn't sure if it would make sense to move this into a shared util or not since it wouldn't be running on the actual site build (and I couldn't export it from the story since that clashes with how Storybook expects exports to work).